lib/jobs/ucs-discovery.js

```diff
@@ -186,15 +199,69 @@ function UcsDiscoveryJobFactory(
         });
     };
 
+    UcsDiscoveryJob.prototype.updateRelations = function(nodeId, relations) {
+        // Update existing node with new relations or create one
+        return waterline.nodes.needOneById(nodeId)
+            .then(function(curNode) {
+                relations = _.uniq(relations.concat(curNode.relations), 'relationType');
+                return waterline.nodes.updateOne(
+                    { id: curNode.id },
+                    { relations: relations }
+                );
+            });
+    };
 
-    /**
-     * @function getRoot
-     * @description create obm setting for the node
-     */
-    UcsDiscoveryJob.prototype.createObms = function (nodeId, obm){
-        return waterline.obms.upsertByNode(nodeId, obm)
-            .then(function(n) {
-                return n;
+
+    UcsDiscoveryJob.prototype.createNode = function(data, type) {
+
+        var self = this;
+        var config = Object.assign({}, self.settings);
+
+        assert.string(data.name);
+        assert.string(data.path);
+
+        var id;
+        if (type === 'compute')
+        {
+            assert.array(data.macs);
+            id = data.macs;
+        } else {
+            id = [config.ucs, data.path];
+        }
+        assert.string(data.name);
+        assert.string(data.path);
+
+        var obm = {
+            config: config,
+            service: 'ucs-obm-service'
+        };
+        var node = {
+            type: type,
+            name: data.path,
+            identifiers: id,
+            obm:[obm],
+            relations : []
+        };
+
+        // Update existing node  or create a new one
+        return waterline.nodes.needOne({identifiers:data.macs})
+            .then(function(curNode) {
+                return waterline.nodes.updateOne(
+                    { id: curNode.id },
+                    node
+                )
+                    .then(function(data){
+                        return data;
+                    });
+            })
+            .catch(function(error) {
+                if (error.name === 'NotFoundError') {
+                    return waterline.nodes.create(node);
+                }
+                throw error;
+            })
+            .then(function(data){
+                return [data, waterline.obms.upsertByNode(data.id,obm)];
```

Comment:
This is not a bug since this function is private and only used by your job, but may not a good design.  This function will return `Promise`, but how should the caller know it should use the `spread` rather than the `then` to get the returned node document? The return value of `waterline.obms.upsertByNode` actually not be used anywhere, so it should be better hidden in this function itself. 

For this comment, either change this or not is OK for me.